app/modules/intelligence/provider/provider_service.py

```
@@ -460,18 +461,25 @@
+        if provider in self._api_key_secrets:
+            return self._api_key_secrets[provider]
+
         env_key = os.getenv("LLM_API_KEY", None)
         if env_key:
+            self._api_key_secrets[provider] = env_key
             return env_key
 
         try:
             secret = SecretManager.get_secret(provider, self.user_id, self.db)
+            self._api_key_secrets[provider] = secret
             return secret
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)